src/box/lua/config/init.lua

```diff
@@ -474,13 +480,26 @@ function methods.reload(self)
     end
 end
 
-function methods.info(self)
+function methods.info(self, version)
     selfcheck(self, 'info')
-    return {
-        alerts = self._aboard:alerts(),
-        meta = self._metadata,
-        status = self._status,
-    }
+    if version == nil or version == 'v1' then
+        return {
+            alerts = self._aboard:alerts(),
+            meta = self._metadata,
+            status = self._status,
+        }
+    end
+    if version == 'v2' then
+        return {
+            alerts = self._aboard:alerts(),
+            meta = {
+                active = self._active_metadata,
+                last = self._metadata,
+            },
+            status = self._status,
+        }
+    end
+    error(('Expected "v2" or nil as info version, got %q'):format(version), 0)
```

Comment:
Nit: The error message is a bit inaccurate. I mean the following.

1. It doesn't mention 'v1'.
2. It requires updating with every version change (and it is easy to overlook).

I would update it this way:

```lua
local function set2string(s)
    local list = {}
    for k, _ in pairs(s) do
        table.insert(list)
    end 
    table.sort(list)
    return ('"%s"'):format(table.concat(list, '", "'))
end

function methods.info(self, version)
    selfcheck(self, 'info')
    version = version or 'v1'
    local supported_versions = { 
        v1 = true,
        v2 = true,
    }   
    if supported_versions[version] == nil then
        error(('config:info() expects nil or %s as an argument, got %q'):format(
            set2string(supported_versions), version), 0)
    end 
    <...>
end
```

Or don't mention versions at all (but this information looks useful).

Or just hardcode `nil, "v1" or "v2"` and try to remember to update it next time.

Reply:
Fixed, rewrote the code.

Reply:
Sorry for nitpicking, but I would highlight two tiny points.

1. `:alerts()` is called twice. It has a cycle inside and it seems easy to avoid it.
2. `pairs(info)` order is arbitrary, so I would add `table.sort(supported_versions)`.

Reply:
Thanks, fixed.